CIScripts/Test/Tests/vRouterAgentMSIInstaller.Tests.ps1

```diff
@@ -1,37 +1,55 @@
 Param (
-    [Parameter(Mandatory=$true)] [string] $TestbedAddr
+    [Parameter(Mandatory=$true)] [string] $TestbedAddr,
+    [Parameter(Mandatory=$true)] [string] $ConfigFile
 )
 
 . $PSScriptRoot\..\Utils\CommonTestCode.ps1
+. $PSScriptRoot\..\Utils\ComponentsInstallation.ps1
 . $PSScriptRoot\..\TestConfigurationUtils.ps1
 . $PSScriptRoot\..\..\Common\Aliases.ps1
 . $PSScriptRoot\..\..\Common\VMUtils.ps1
 . $PSScriptRoot\..\PesterHelpers\PesterHelpers.ps1
 
-. $PSScriptRoot\..\GetTestConfigurationCodiLegacy.ps1
+. $ConfigFile
 $TestConf = Get-TestConfiguration
 $Session = New-PSSession -ComputerName $TestbedAddr -Credential (Get-VMCreds)
 
 Describe "vRouter Agent MSI installer" {
-    Context "installation and uninstallation" {
-        It "registers/unregisters agent service and never enables" {
-            Install-Agent -Session $Session
-            Eventually {
-                Get-AgentServiceStatus -Session $Session | Should Be "Stopped"
-            } -Duration 3
-
-            Uninstall-Agent -Session $Session
-            Eventually {
-                {Get-AgentServiceStatus -Session $Session} | Should Throw
-            } -Duration 3
+
+    function Test-AgentMSIBehaviourCorrect {
+        Install-Agent -Session $Session
+        Eventually {
+            Get-AgentServiceStatus -Session $Session | Should Be "Stopped"
+        } -Duration 3
+
+        Uninstall-Agent -Session $Session
+        Eventually {
+            Get-AgentServiceStatus -Session $Session | Should Throw
+        } -Duration 3
+    }
+
+    Context "when vRouter Forwarding Extension is not running" {
+        It "registers/unregisters Agent service and never enables" {
+            Test-AgentMSIBehaviourCorrect
         }
     }
 
-    BeforeEach {
-        # TODO Install and enable Docker Driver in this test,
-        # so we can actually check that Agent doesn't auto-start
-        # (this is commented out because docker driver doesn't build yet)
-        # Initialize-DriverAndExtension -Session $Session -TestConfiguration $TestConf
+    Context "when vRouter Forwarding Extension is running" {
+        It "registers/unregisters Agent service and never enables" {
+            Test-AgentMSIBehaviourCorrect
+        }
+
+        BeforeEach {
+            Install-Extension -Session $Session
+            Enable-VRouterExtension -Session $Session -AdapterName $TestConf.AdapterName `
+                -VMSwitchName $TestConf.VMSwitchName `
+                -ForwardingExtensionName $TestConf.ForwardingExtensionName
```

Comment:
Do we need those parameters to `Enable-VRouterExtension`? We always call it with the same set of params.

Reply:
Eventual fix is to have a `Testbed` class, that would store stuff like name of VM switch as a field. But not in scope of this PR.

Reply:
Well, we kind of have this class already, and it's called `TestConfiguration`. But it's fine not to change it in this PR.